helper/resource/error.go

```diff
@@ -0,0 +1,21 @@
+package resource
+
+type NotFoundError struct {
+	LastError    error
+	LastRequest  interface{}
+	LastResponse interface{}
+	Message      string
+	Retries      int
+}
+
+func (e *NotFoundError) Error() string {
+	if e.Message != "" {
+		return e.Message
+	}
+
+	return "couldn't find resource"
+}
+
+func NewNotFoundError(err string) *NotFoundError {
```

Comment:
I'd like to see **this specific error** being stuffed with as much details as possible in as many places as possible (to make debugging easier) and any helper/wrapper functions like that may discourage from this.

That's why I'm inclined to remove the this completely and let everyone use the `&NotFoundError{}` format as I'd **want** them to add more than just an error message.

Something like `&NotFoundError{Message: err}` should naturally ring an alarm bell for the reviewer who should be asking _"can we capture more details?"_ :wink: 

Function with one or two arguments would make you feel that interface was satisfied, so it's ok.

This may not be true for other custom errors (like `RetryError`), but IMO is true for `NotFoundError`.